### PR TITLE
323 navbar responsive

### LINKED ISSUE #323: Make the landing page Navbar responsive
**Is your feature request related to a problem? Please describe.**
Right now the landing page is not responsive and it is not looking good in mobile devices 

**Describe the solution you'd like**
make the Navbar responsive and add a dropdown for the calculators 

**Describe alternatives you've considered**
NA

**Additional context**
NA

### PR DESCRIPTION
**Pull request by imhardikdesai**
### Description
This pull request addresses the issue of the responsive navbar not displaying correctly on various screen sizes. The changes made in this PR aim to make the navbar more responsive and visually appealing on all devices.

### Changes Made
- Updated CSS styles to ensure that the navbar elements adapt well to different screen sizes.
- Take a Reference from `https://refine.dev/` as @mohitkhatri06 suggested
- Implemented a mobile-friendly menu for small screens.
- Ensured that the navigation links are easily accessible and functional on all devices.

### Screenshots 
image
image
image

### Issues Resolved
Closes #323

Please review this pull request, and if there are any suggestions or changes needed, feel free to let me know. Your feedback is greatly appreciated.

### CODE REVIEW

**Review by bharathkumarchowdary — DISMISSED** [review]

File: package.json [hunk 1]
```diff
@@ -11,6 +11,7 @@
     "@radix-ui/react-dialog": "^1.0.5",
     "@radix-ui/react-dropdown-menu": "^2.0.6",
     "@radix-ui/react-label": "^2.0.2",
+    "@radix-ui/react-navigation-menu": "^1.1.4",
```
Comment (rockingrohit9639) [inline package.json:14]: We don't have to install this dependency. We already have `Drawer` installed.. we can make use of that component to create a slider and that will be visible on mobile.

File: src/components/AppShell/components/NavLink.tsx [hunk 2]
```diff
@@ -6,16 +6,17 @@ import { cn } from '@/utils/utils'
...
-export default function NavLink({ to, icon, label }: Props) {
+export default function NavLink({ to, icon, label, twClass }: Props) {
   const location = useLocation()
   const isActive = location.pathname.startsWith(to as string)
 
   return (
     <Link
       to={to}
-      className={cn(buttonVariants({ variant: isActive ? 'default' : 'ghost' }), 'w-full justify-start gap-2')}
+      className={cn(buttonVariants({ variant: (isActive ? 'default' : 'ghost') }), 'w-full justify-start gap-2', twClass)}
```
Comment (rockingrohit9639) [inline src/components/AppShell/components/NavLink.tsx:19]: Unnecessary `twClass`
We are extending props with LinkProps so you don't have to declare another props just for getting the class... you can directly get it with `className` prop

File: src/components/UnAuthenticatedHeader/UnAuthenticatedHeader.tsx [hunk 3]
```diff
@@ -21,31 +56,101 @@ export default function UnAuthenticatedHeader() {
...
+         {/* Hamburger icon for mobile menu */}
+         <div className='md:hidden' onClick={handleMobileMenuToggle}>
+          <div className='w-6 h-6 cursor-pointer'>
+            {
+              isMobileMenuOpen ? <X /> : <Menu />
+            }
+          </div>
+        </div>
       </div>
     </div>
```
Comment (rockingrohit9639) [inline src/components/UnAuthenticatedHeader/UnAuthenticatedHeader.tsx:155]: Update this file according to `Drawer`

File: src/components/ui/navigation-menu.tsx [hunk 4]
```diff
@@ -0,0 +1,128 @@
...
+  navigationMenuTriggerStyle,
+  NavigationMenu,
+  NavigationMenuList,
+  NavigationMenuItem,
+  NavigationMenuContent,
+  NavigationMenuTrigger,
+  NavigationMenuLink,
+  NavigationMenuIndicator,
+  NavigationMenuViewport,
+}
```
Comment (rockingrohit9639) [inline src/components/ui/navigation-menu.tsx:128]: Remove this file.

File: package.json (hunk 1 above)
Comment (mohitkhatri06) [reply package.json:14]: @rockingrohit9639 for the desktop, I suggested to have a menu on the hover on the calculator

**Review by mohitkhatri06 — APPROVED** [review]

**Pull request merged**